### PR TITLE
add RAG DA scripts

### PR DESCRIPTION
**Pull request by rajatagarwal-ibm**
### Description

prevalidation and postvalidation scripts 

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Review by brendankellyibm — DISMISSED** [review]

File: ibm_catalog.json [hunk 1]
```diff
@@ -73,11 +26,8 @@
               {
                 "profile": {
                   "name": "IBM Cloud for Financial Services",
-                  "version": "1.4.0"
-                },
-                "names": [
-                  "SA-4(b)"
-                ]
+                  "version": "1.6.0"
+                }
```
Comment (ocofaigh) [inline ibm_catalog.json:30]: this syntax is wrong - it should be:
```
"compliance": {
                        "authority": "scc-v3",
                        "profiles": [
                            {
                                "profile_name": "IBM Cloud Framework for Financial Services",
                                "profile_version": "1.6.0"
                            }
                        ]
                    }
```

**Review by ocofaigh — CHANGES_REQUESTED** [review]
1 comment

**Review by ocofaigh — APPROVED** [review]

**Comment by ocofaigh** [comment]
@brendankellyibm @rajatagarwal-ibm tests failed with:
```
│ Error: CreateTektonPipelineWithContext failed Forbidden
│ {
│     "StatusCode": 403,
│     "Headers": {
│         "Cf-Cache-Status": [
│             "DYNAMIC"
│         ],
│         "Cf-Ray": [
│             "872252a709de7fdc-IAD"
│         ],
│         "Content-Security-Policy": [
│             "default-src 'self';style-src 'self' https://*.liveperson.net https://*.lpsnmedia.net https://cdn.walkme.com https://*.medallia.eu https://*.kampyle.com https://cloud.ibm.com 'unsafe-inline';script-src 'self' 'unsafe-eval' https://cdn.segment.com https://*.prefmgr-cookie.truste-svc.net https://consent.truste.com https://consent.trustarc.com https://*.go-mpulse.net https://*.walkme.com https://*.market-pay.com https://*.marketplace.ibm.com https://*.liveperson.net https://*.lpsnmedia.net https://*.kampyle.com https://*.medallia.eu https://cloud.ibm.com 'unsafe-inline';script-src-attr 'self' https://*.liveperson.net https://*.lpsnmedia.net 'nonce-dcbbfbd60eeda019aa865e39e72034d1';connect-src 'self' https://api.segment.io https://*.akstat.io https://prefmgr-cookie.truste-svc.net https://*.trustarc.com https://*.walkme.com https://*.go-mpulse.net https://cdn.segment.com https://*.ice.ibmcloud.com wss://va.msg.liveperson.net https://*.liveperson.net https://*.softlayer.com 'nonce-dcbbfbd60eeda019aa865e39e72034d1' https://*.medallia.eu https://*.kampyle.com https://cloud.ibm.com https://login.ibm.com https://*.softlayer.com https://*.akamaihd.net;child-src 'self' https://consent.trustarc.com https://*.medallia.eu https://*.kampyle.com;font-src 'self' data: https://*.s81c.com https://*.medallia.eu https://*.kampyle.com;base-uri 'self' 'nonce-dcbbfbd60eeda019aa865e39e72034d1' https://*.kampyle.com;form-action 'self';frame-ancestors 'self';img-src 'self' data: https: https://*.lpsnmedia.net https://*.medallia.eu https://*.kampyle.com;media-src 'self' https://*.ibm.com https://*.bluemix.net https://*.containers.appdomain.cloud https://*.lpsnmedia.net;frame-src 'self' https://*.trustarc.com https://*.market-pay.com https://*.marketplace.ibm.com https://*.liveperson.net https://*.lpsnmedia.net https://*.medallia.eu https://*.kampyle.com;report-uri /devops/pipelines/csp-report?env_id=ibm:yp:us-south;object-src 'none';upgrade-insecure-requests"
│         ],
│         "Content-Type": [
│             "application/json; charset=utf-8"
│         ],
│         "Date": [
│             "Wed, 10 Apr 2024 11:16:07 GMT"
│         ],
│         "Etag": [
│             "W/\"9f-ti6vpkWINugLdWvRwXaTOfj+CuU\""
│         ],
│         ],
│         "Content-Type": [
│             "application/json; charset=utf-8"
│         ],
│         "Date": [
│             "Wed, 10 Apr 2024 11:16:07 GMT"
│         ],
│         "Etag": [
│             "W/\"9f-NJnHm9ZT+1aU9wCQua/+7UXPV8Y\""
│         ],
│         "Referrer-Policy": [
│             "no-referrer-when-downgrade"
│         ],
│         "Server": [
│             "cloudflare"
│         ],
│         "Strict-Transport-Security": [
│             "max-age=31[536](https://github.com/terraform-ibm-modules/terraform-ibm-rag-sample-da/actions/runs/8630241484/job/23656264167#step:7:537)000; includeSubDomains"
│         ],
│         "Vary": [
│             "Accept-Encoding"
│         ],
│         "X-Content-Type-Options": [
│             "nosniff"
│         ],
│         "X-Envoy-Upstream-Service-Time": [
│             "311"
│         ]
│     },
│     "Result": {
│         "errors": [
│             {
│                 "code": 403,
│                 "message": "Continuous Delivery service required"
│             }
│         ],
│         "trace": "18a970d8-561b-4301-9ff4-d2f49b9fe0fd"
│     },
│     "RawResult": null
│ }
│ 
│ 
│   with ibm_cd_tekton_pipeline.cd_tekton_pipeline_instance,
│   on main.tf line 75, in resource "ibm_cd_tekton_pipeline" "cd_tekton_pipeline_instance":
│   75: resource "ibm_cd_tekton_pipeline" "cd_tekton_pipeline_instance" {
│ 
╵}
```

**Comment by brendankellyibm** [comment]
> @rajatagarwal-ibm @brendankellyibm It looks like the [Continuous Delivery](https://cloud.ibm.com/catalog/services/continuous-delivery) service is required to provision the resources in the example being run. I suggest you update the example to create it so tests can pass

Makes sense - I don't think the ALM DA creates the continuous delivery service instance, it assumes we have it. I previously had it in my sample WatsonX SaaS DA, but it doesn't make any sense to put it there. I'll provision it in the app DA.

**Comment by rajatagarwal-ibm** [comment]
@brendankellyibm I have added that to the tests.

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**